### ISSUE TITLE
JTree: FlatLaf does not like components shared between the renderer and the editor

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#385]
In a JTree, when a component is shared between the renderer and the editor, FlatLaf does not show the component in the editor. Windows L&F does not have this problem:
![TreeEditorRendererSharedComponent](https://user-images.githubusercontent.com/1174202/133409200-23700d51-2713-4ca8-b73e-c7165719e5da.png)

What seems to happen, for a given cell:
- The editor is activated, the component is placed in the editor (and thus removed from the renderer).
- The renderer of that cell is called, even though the cell is being edited, which removes the component from the active editor.

I don't know if it is legal to share a component between the editor and the renderer, but I am sure I have seen this more than once. Also, this is in an old application which I wouldn't want to rewrite!

Here is a simple test case, simply switch between look and feels:
[TreeWithEditorAndRendererSharedComponentTest.zip](https://github.com/JFormDesigner/FlatLaf/files/7168878/TreeWithEditorAndRendererSharedComponentTest.zip)

Please let me know what you think, and if something can be done to support this case 😉

**Comment by DevCharly** [comment]
Looks like a bug.
The tree cell renderer of the editing cell should not be used...

Does this also happen for JTable?
The screenshots and the example are for JTree, but the issue title/text mentions JTable.

**Comment by Chrriis** [comment]
Out of curiosity, I have tried the same for JTable, and it works as expected. Here is my test case:
[TableWithEditorAndRendererSharedComponentTest.zip](https://github.com/JFormDesigner/FlatLaf/files/7170390/TableWithEditorAndRendererSharedComponentTest.zip)

So, the problem seems to only affect JTree.

**Comment by DevCharly** [comment]
There is a workaround: disable selection painting for the tree.

~~~java
tree.putClientProperty( FlatClientProperties.TREE_PAINT_SELECTION, false );
~~~

Then the test case works.

**Comment by DevCharly** [comment]
The fix for this issue is now in FlatLaf 1.6.3.
